### PR TITLE
fix : adding support for contains operator

### PR DESCRIPTION
**Pull request by kotharironak**
## Description
Adds support for contains operator on Postgres implementation.

### Testing
- Added basic unit test,
- working on adding integration test.

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by kotharironak** [comment]
@suresh-prakash I have added the integration test. we will refactor after discussing. But, this will add functional support for CONTAINS in both mongo + postgres.

File: document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/parser/MongoRelationalExpressionParser.java [hunk 1]
```diff
@@ -91,6 +99,16 @@ private static Map<String, Object> generateMap(
     return (lhs, rhs) -> {
       final String parsedLhs = lhs.accept(identifierParser);
       final Object parsedRhs = rhs.accept(rhsParser);
+      Map<String, String> parsedObject = Map.of();
+      if (parsedRhs instanceof String) {
+        final ObjectReader objectReader = new ObjectMapper().readerFor(Map.class);
```
Comment (suresh-prakash) [inline document-store/src/main/java/org/hypertrace/core/documentstore/mongo/query/parser/MongoRelationalExpressionParser.java:104]: This logic can be as simple as we have done in `org.hypertrace.core.documentstore.mongo.subdoc.MongoSubDocumentValueParser.parse`
That comes with necessary sanitization, etc.

File: document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java [hunk 2]
```diff
@@ -102,4 +114,25 @@ private Collector getCollectorForLogicalOperator(LogicalOperator operator) {
...
+    switch (operator) {
+      case CONTAINS:
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  private Object convertStringValueToMap(Object value) {
+    if (value instanceof String) {
```
Comment (suresh-prakash) [inline document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java:128]: Can we do this a bit cleanly?
Maybe, ConstantExpression can take in a document and handle the case.
Please refer to the way it is done in `NestedSubDocumentValue`.
Basically, we should be able to differentiate a literal String vs a JSON String.

File: document-store/src/main/java/org/hypertrace/core/documentstore/expression/operators/RelationalOperator.java [hunk 3]
```diff
@@ -12,6 +12,7 @@ public enum RelationalOperator {
   LTE("<="),
   IN("IN"),
   CONTAINS("CONTAINS"),
+  NOT_CONTAINS("NOT_CONTAINS"),
```
Comment (suresh-prakash) [inline document-store/src/main/java/org/hypertrace/core/documentstore/expression/operators/RelationalOperator.java:15]:
```suggestion
  NOT_CONTAINS("NOT CONTAINS"),
```
to be in sync. with "NOT IN" and "NOT EXISTS".
But, not a blocker and can come in a separate PR too.

File: document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java [hunk 4]
```diff
@@ -102,4 +113,26 @@ private Collector getCollectorForLogicalOperator(LogicalOperator operator) {
...
+    switch (operator) {
+      case CONTAINS:
+      case NOT_CONTAINS:
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  private Object convertStringValueToMap(Object value) {
```
Comment (suresh-prakash) [inline document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java:127]: This method can be removed.

**Review by suresh-prakash — DISMISSED** [review]
All are non-blocking comments. Can do them separately in a follow-up.

File: document-store/src/main/java/org/hypertrace/core/documentstore/expression/operators/RelationalOperator.java (hunk 3 above)
Comment (kotharironak) [reply document-store/src/main/java/org/hypertrace/core/documentstore/expression/operators/RelationalOperator.java]: Done

**Review by suresh-prakash — DISMISSED** [review]

**Pull request merged**